.pre-commit-config.yaml.failing

```diff
@@ -3,6 +3,7 @@ repos:
   rev: v1.9.1
   hooks:
   - id: pylint
+    language_version: python2.7
```

Comment:
I'd want it to run it under Python 3.7 or any currently active python.

Could you rewrite it like this? https://github.com/sanitizers/chronographer-github-app/blob/master/.pre-commit-config.yaml#L50-L59
pylint is a special case: it makes not only static but also semi-dynamic checks like checking whether things are imporatable which is impossible from a separate isolated virtualenv pre-commit tool creates.

Reply:
I just found that they added compatibility for the 3.7 in pytest 3.0.1, I'm testing to see if it will work back to 2.7. If not, I'll look into using that commit :)

Reply:
Well, the version of pytest used currently is 4 so it must be ok

Reply:
That's what I'm curious about. I just noticed it was running on 4, which is way past the [commit](https://docs.pytest.org/en/latest/changelog.html#id553) where they fixed the issue, yet there is still a StopIteration laying around in the source.

Reply:
But regarding commit messages: you could improve wording there + it looks like you're committing from unconfigured computer. The email you use there doesn't match one bound to your GitHub account, that's why they ain't linked to you.

Reply:
Oh.. And it doesn't fail hard in travis because it still uses Python 3.6 for linter jobs...

Reply:
Or not. It's interesting: this fails during tests collection but not in Travis (even under Python 3.7). Can it be Travis CI cache using different version of (indirect?) dependency for some pytest plugin, not pytest itselff?

Reply:
Huh, check that out. I'll work on that and push out a fix. I'd really like to have the tests be useful again, especially for my own purposes.

Reply:
Sure, thanks!

It looks like this line is causing `StopIteration` being raised: https://github.com/pytest-dev/pytest/blob/master/src/_pytest/main.py#L586. Which probably means that object being empty:
```python
>>> next(iter([]))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
StopIteration
```

This may be caused by `testmon` plugin if it's active. Or... Maybe it's because CircleCI's CLI command returns empty list (it's used to distribute tests across 4 parallel jobs).

P.S. We mostly rely on Travis CI currently so broken Circle CI is not a major issue.

Reply:
Oh, there's `--testmon-off` so it's definitely not it, only Circle CI client left to be checked.

Reply:
It's probably related to env in that docker image used in Circle CI. You can test it locally.